Borders: add design callouts messages (#2064)
Orange-OpenSource/Orange-Boosted-Bootstrap: site/content/docs/5.3/utilities/borders.md

Or modify the default `border-color` of a component:

{{< design-callout-alert >}}

> These border color variants, which are just **examples illustrating the use of this color utility**, should not be used because they do not respect the Orange Design System specifications.

Reviewer: ```suggestion
These border color variants are just **examples illustrating the use of this color utility**. They should not be used because they do not respect the Orange Design System specifications.
```

Author: Indeed, you proposal is "lighter" in style. If we change this message, then we should do the same for both these messages formulated the same way : 
- Components/List-group.md/#horizontal
- Components/nav-tabs.md/#fill&justify

Reviewer 2: Prefer the Julien proposition since it uses simple sentences.

Reviewer: Modified in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/2064/commits/4b60154da0b6b709c8dcc2338112e99c0d8381dc

Or modify the default `border-color` of a component:

{{< design-callout-alert >}}

> The **second and third** border color variants should not be used because they do not respect the Orange Design System specifications.

Reviewer 2: ```suggestion
The **second and third** border color variants should not be used in this context because they do not respect the Orange Design System specifications.
```

Author: Result of the discussion on Teams / ODS-WEB-Boosted : 
Julien + LM : IMO it avoids confusion to let the color coming from Bootstrap because using red might be OK in terms of rendering, but the developers won't never apply the border utility manually on an input (can be misleading here). On top of that, still IMO, it's easier to maintain, and easier to explain in the callout message since all the examples are not compilant with the DSM (can be explained more easily).

IC : OK, then let's not change the email address input border for red (we keep the green border coming from Bootstrap) and the design callout message will concern the 3 examples.
=> I'm going to change the design callout message

## Width

{{< design-callout-alert >}}

> The border variant with **a width of 3px** should not be used because it does not respect the Orange Design System specifications.

Reviewer: ```suggestion
The **`.border-3` variant** should not be used because it does not respect the Orange Design System specifications.
```

Author: OK for your proposal. My message is formulated like several other design callout messages where some of the illustrated variants shall be indicated.

Reviewer: Haven't seen it before but there are 5 variants in Bootstrap (see https://twbs-bootstrap.netlify.app/docs/5.3/utilities/borders/#width), so we should reintegrate the 5 and modify the callout to say that 3, 4 and 5 shouldn't be used.
Do you mind if I add a commit to reintegrate these border utilities?

Reviewer: Done via https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/2127

Reviewer 2: ```suggestion
The **`.border-{3|4|5}` variants** should not be used because it does not respect the Orange Design System specifications.
```

> The three last border utilities with **a width superior to 2px** should not be used because they do not respect the Orange Design System specifications.

Author: OK for the modification. I just would replace "a width superior to 2px" by one of these 3 propositions:
- "a width greater than 2px" 
- "a width larger than 2px"
- "a width of more than 2px"

> The three last border utilities with **a width superior to 2px** should not be used because they do not respect the Orange Design System specifications.

Reviewer: ```suggestion
The three last border utilities with **a width larger than 2px** should not be used because they do not respect the Orange Design System specifications.
```

> The three last border utilities with **a width superior to 2px** should not be used because they don't not respect the Orange Design System specifications.

Reviewer: ```suggestion
The three last border utilities with **a width superior to 2px** should not be used because they do not respect the Orange Design System specifications.
```

## Width

{{< design-callout-alert >}}

> These border variants with **a width of 0px, 1px and 3px** should not be used because they do not respect the Orange Design System specifications.

Reviewer 2: ```suggestion
The border variant with **a width of 3px** should not be used because they do not respect the Orange Design System specifications.
```
I'd rather make it this way

Add classes to an element to easily round its corners.

{{< design-callout-alert >}}

> These border variants with **a radius css class different than `rounded` and `rounded-pill`** should not be used because they do not respect the Orange Design System specifications.

Reviewer: ```suggestion
These border variants **other than `.rounded`, `.rounded-pill` and `.rounded-circle`** should not be used because they do not respect the Orange Design System specifications.
```

Note: `.rounded-circle` could be used to render stickers or other rounded elements present in the DSM.

Author: I like my version better. I can add a `.`before the class names. I would be:
> These border variants with **a radius css class different than `.rounded` and `.rounded-pill`** should not be used because they do not respect the Orange Design System specifications.

What do like better @louismaximepiton, @hannahiss, @MewenLeHo?

Reviewer 2: Can we use `.rounded`? I think we can only use `.rounded-pill` and `.rounded-circle`?
On the syntax side, I'd prefer the dot syntax for classes for sure, once done, I think there's no reason to repeat ourselves around the classes. The simpler the better ?

Reviewer: OK so I kept the original sentence, added the dot for the classes, changed "css" to "CSS". Regarding `.rounded`, it's never used and some rounded elements in the DSM don't use exactly the same value as the `.rounded` one; I'm thinking for instance about the countdown in https://system.design.orange.com/0c1af118d/p/50e873-e-shop/b/046123.
See https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/2064/commits/1a18802429789b0774dcf36fde0a64c7fa268783.

Use the scaling classes for larger or smaller rounded corners. Sizes range from `0` to `5`, and can be configured by modifying the utilities API.

{{< design-callout-alert >}}

> These border variants with **a size css class different than `rounded-0`** should not be used because they do not respect the Orange Design System specifications.

Reviewer: ```suggestion
These border variants **other than `.rounded-0`** should not be used because they do not respect the Orange Design System specifications.
```

Reviewer: OK kept the original sentence, modified "css" to "CSS", and added the dot in front of the classes. See https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/2064/commits/73ea34c925f4cb946c15a6b64838d227cf739040.
